src/contrib/dependencyinjection/Akka.DependencyInjection.Tests/ActorServiceProviderPropsWithScopesSpecs.cs

```diff
@@ -0,0 +1,199 @@
+        {
+            var spExtension = ServiceProvider.For(Sys);
+            var props = spExtension.Props(sp => new ScopedActor(sp.CreateScope()));
+
+            // create a scoped actor using the props from Akka.DependencyInjection
+            var scoped1 = Sys.ActorOf(props, "scoped1");
+            scoped1.Tell(new FetchDependencies());
+            var deps1 = ExpectMsg<CurrentDependencies>();
+            deps1.Dependencies.All(x => x.Disposed).Should().BeFalse();
+
+            // kill it
+            Watch(scoped1);
+            scoped1.Tell(PoisonPill.Instance);
+            ExpectTerminated(scoped1);
+
+            // all dependencies should be disposed
+            deps1.Dependencies.All(x => x.Disposed).Should().BeTrue();
+
+            // reuse the same props
+            var scoped2 = Sys.ActorOf(props, "scoped2");
+            scoped2.Tell(new FetchDependencies());
+            var deps2 = ExpectMsg<CurrentDependencies>();
+
+            // all dependencies should be current, because new scope was generated by Props
+            deps2.Dependencies.All(x => x.Disposed).Should().BeFalse();
+        }
+
+        [Fact(DisplayName =
+            "DI: actors who receive an IServiceScope through Props should dispose of their dependencies and recreate upon restart")]
+        public void ActorsWithScopedDependenciesShouldDisposeAndRecreateUponRestart()
+        {
+            var spExtension = ServiceProvider.For(Sys);
+            var props = spExtension.Props(sp => new ScopedActor(sp.CreateScope()));
+
+            // create a scoped actor using the props from Akka.DependencyInjection
+            var scoped1 = Sys.ActorOf(props, "scoped1");
+            scoped1.Tell(new FetchDependencies());
+            var deps1 = ExpectMsg<CurrentDependencies>();
+            deps1.Dependencies.All(x => x.Disposed).Should().BeFalse();
+
+            // crash + restart it
+            EventFilter.Exception<ApplicationException>().ExpectOne(() =>
+            {
+                scoped1.Tell(new Crash());
+            });
+
+            // all previous dependencies should be disposed
+            deps1.Dependencies.All(x => x.Disposed).Should().BeTrue();
+
+            // actor should restart with totally new dependencies
+            scoped1.Tell(new FetchDependencies());
+            var deps2 = ExpectMsg<CurrentDependencies>();
+            deps2.Dependencies.All(x => x.Disposed).Should().BeFalse();
+        }
+
+        [Fact(DisplayName =
+            "DI: actors who receive a mix of dependencies via IServiceScope should dispose ONLY of their scoped dependencies and recreate upon restart")]
+        public void ActorsWithMixedDependenciesShouldDisposeAndRecreateScopedUponRestart()
+        {
+            var spExtension = ServiceProvider.For(Sys);
+            var props = spExtension.Props(sp => new MixedActor(sp.GetRequiredService<AkkaDiFixture.ISingletonDependency>(), sp.CreateScope()));
```

Comment:
An example of mitching and maxing DI scopes here - note that the container _will_ do this for you since the dependencies are scoped properly inside our class fixture, but I wanted an example inside the test suite.

Note: you could also pass in an closed-over argument here that would be included inside the `Props` for the actor when it's created. This is how you can mix and match DI easily.

Theoretically, using this method, you could also create the actor instance _itself_ with DI:

```
spExtension.Props(sp => sp.GetService<MixedActor>());
```

That would also work.

Reply:
```spExtension.Props(sp => sp.GetService<MixedActor>());``` will not work, because MixedActor is not registered as a transient service into the ServiceCollection

This is the why to go und should have some kind of shorter support:
```spExtension.Props(sp => ActivatorUtilities.CreateInstance<MixedActor>(sp, "extraParam1", "extraParam2"));```

Props have a Arguments array that can me used for the extra params to make this like the already overload ```spExtension.Props<MixedActor>("extraParam1", "extraParam2");```
Props.Arguments would then hold the extra params

The Props should then call following method to create the actor instance internal:
```csharp
public override ActorBase NewActor() { 
   //...
   var instance = (ActorBase)ActivatorUtilities.CreateInstance(sp, Type, Arguments);
   //...
   return instance;
}
```

In the simplest form the ```Props.Arguments``` are empty.

Maybe the SP-Props could replace Normal-Props as default actor factory (configurable).

An other long standing question for me is the lifetime of the Props instance itself.
There is a small logical Problem. A Props-Instance hold/reference all Parameters  
for the lifetime of the actor. It can be accessed by ```Context.Props.Arguments```
But all example show ````Props.Create(() => new MyActor(param1, param2))``` 
that would create an capture-scope and delegate loaded for the whole life of the actor,
even with a Stopping Strategy (do not restart)

Reply:
I like it - I'll add it to this PR.

The `MixedActor` sample will work, however, as the service we're injecting via a manual call is a singleton so having a registered service isn't necessary in this instance, but your point is taken.

Reply:
So I just updated this in my most recent commit - problem: actors created using `ActivatorUtilities` and DI container registration still have to dispose of their own dependencies, by the looks of things. Had to implement a manual `IDisposable` method to get the test to pass.